### PR TITLE
Gives janicart a radial menu

### PR DESCRIPTION
**Pull request by Henri215**
## What Does This PR Do

title
## Why It's Good For The Game

Radial menus make it easier to grab something from the janicart without having to open a whole new window.
## Images of changes

image

## Testing

- Inserted and removed all possible items from it
## Changelog
:cl:
tweak: Janicart now uses a radial menu
/:cl:

### CODE REVIEW

**Comment by Inferno1882** [comment]
Can we also expand this to the EngiCart?

**Comment by Henri215** [comment]
> Can we also expand this to the EngiCart?

if they could use a radial menu then yeah, but in another PR.

File: code/game/objects/structures/janicart.dm [hunk 1]
```diff
@@ -176,7 +181,13 @@
...
 
 	update_icon(UPDATE_OVERLAYS)
-	updateUsrDialog()
+
+/obj/structure/janitorialcart/proc/check_menu(mob/living/user)
+	if(!istype(user))
+		return FALSE
+	if(user.incapacitated())
+		return FALSE
+	return TRUE
```
Comment (Contrabang) [inline code/game/objects/structures/janicart.dm:190]:
```suggestion
	return (istype(user) && !HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
```

**Review by Contrabang — APPROVED** [review]
LGTM, CI seems to be randomly failing, probably just needs to be rerun

**Review by S34NW — APPROVED** [review]

**Pull request merged**